core/indigo-core/molecule/src/molecule_json_loader.cpp

```diff
@@ -956,6 +973,204 @@ void MoleculeJsonLoader::parseProperties(const rapidjson::Value& props, BaseMole
     }
 }
 
+void MoleculeJsonLoader::fillXBondsAndBrackets(Superatom& sa, BaseMolecule& mol)
+{
+    std::unordered_set<int> atoms;
+    std::vector<Vec2f> brackets;
+    for (auto val : sa.atoms)
+        atoms.insert(val);
+
+    // fill crossing bonds
+
+    for (auto src_atom : sa.atoms)
+    {
+        const auto& vx = mol.getVertex(src_atom);
+        const auto& src_pos = mol.getAtomXyz(src_atom);
+        for (int i = vx.neiBegin(); i != vx.neiEnd(); i = vx.neiNext(i))
+        {
+            auto target_atom = vx.neiVertex(i);
+            if (atoms.find(target_atom) == atoms.end())
+            {
+                const auto& target_pos = mol.getAtomXyz(target_atom);
+                sa.bonds.push(vx.neiEdge(i));
+                brackets.emplace_back((target_pos.x - src_pos.x) / 2, (target_pos.y - src_pos.y) / 2);
+            }
+        }
+    }
+
+    // fill brackets
+
+    for (int i = 0; i < brackets.size(); i += 2)
```

Comment:
Looks strange.
Is it really each bracket should have second member set to 0 if brackets count is odd?
May be it should be only for last bracket?

Reply:
fixed